File: AndBible/src/net/bible/android/control/speak/NumPagesToSpeakDefinition.java
package net.bible.android.control.speak;

import net.bible.android.BibleApplication;

public class NumPagesToSpeakDefinition {
	private int numPages;
	private int resourceId;
	private boolean isPlural;
	private int radioButtonId;
	
	public NumPagesToSpeakDefinition(int numPages, int resourceId, boolean isPlural, int radioButtonId) {
		super();
		this.numPages = numPages;
		this.resourceId = resourceId;
		this.isPlural = isPlural;
		this.radioButtonId = radioButtonId;
	}
	
	public String getPrompt() {
		String prompt = null;
		if (isPlural) {
			prompt = BibleApplication.getApplication().getResources().getQuantityString(resourceId, numPages, numPages);
		} else {
			prompt = BibleApplication.getApplication().getResources().getString(resourceId);
		}
		return prompt;
	}
	
	public int getRadioButtonId() {
		return radioButtonId;
	}

	public void setNumPages(int numPages) {
		this.numPages = numPages;
	}

	public int getNumPages() {
		return numPages;
	}
}

File: app/src/main/java/net/bible/android/control/speak/NumPagesToSpeakDefinition.kt
package net.bible.android.control.speak

import net.bible.android.BibleApplication

/**
 * Support the Number of pages to Speak prompt
 *
 * @author Martin Denham [mjdenham at gmail dot com]
 * *
 * @see gnu.lgpl.License for license details.<br></br>
 * The copyright to this program is held by it's author.
 */
class NumPagesToSpeakDefinition(var numPages: Int, private val resourceId: Int, private val isPlural: Boolean, val radioButtonId: Int) {

    fun getPrompt() =
            if (isPlural) {
                BibleApplication.getApplication().resources.getQuantityString(resourceId, numPages, numPages)
            } else {
                BibleApplication.getApplication().resources.getString(resourceId)
            }
}
